File: src/main/java/cc/carm/plugin/intellij/quarkdown/QuarkdownFileType.java
package cc.carm.plugin.intellij.quarkdown;

import com.intellij.openapi.fileTypes.FileType;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public final class QuarkdownFileType implements FileType {

    public static final QuarkdownFileType INSTANCE = new QuarkdownFileType();

    @NotNull
    @Override
    public String getName() {
        return "Quarkdown";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Quarkdown document";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "qd";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return QuarkdownIcons.FILE;
    }

    @Override
    public boolean isBinary() {
        return false;
    }

    @Override
    public boolean isReadOnly() {
        return false;
    }

    @Nullable
    @Override
    public String getCharset(@NotNull VirtualFile file, @NotNull byte[] content) {
        return null;
    }
}

File: src/main/kotlin/cc/carm/plugin/intellij/quarkdown/QuarkdownFileType.kt
package cc.carm.plugin.intellij.quarkdown

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

class QuarkdownFileType : LanguageFileType(QuarkdownLanguage.INSTANCE) {

    override fun getName(): String = "Quarkdown"

    override fun getDescription(): String =
        QuarkdownBundle.message("quarkdown.filetype.description")

    override fun getDefaultExtension(): String = "qd"

    override fun getIcon(): Icon = QuarkdownIcons.FILE

    override fun isReadOnly(): Boolean = false

    companion object {
        @JvmField
        val INSTANCE = QuarkdownFileType()
    }
}
